fabric-bow-api-v1/src/main/java/net/fabricmc/fabric/mixin/bow/EntityUseBowMixin.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.bow;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.entity.mob.AbstractSkeletonEntity;
+import net.minecraft.item.BowItem;
+import net.minecraft.item.RangedWeaponItem;
+
+// {} is to show that this mixin can be used for many entities
+@Mixin({AbstractSkeletonEntity.class})
```

Comment:
correct me if I'm wrong but don't illusioners also use bows?

Reply:
It's in `BowAttackGoal`, which I think I chose not to modify (not sure why). There are a couple of methods in there that I need to modify, but it's not too major.

The skeleton entities have their own special canUseRangedWeapon method that the illusioner does not. I think this comes from the fact that some entities can hold different items/have it break. Because the illusioner is not finished mojang left some mojank.

Reply:
`BowAttackGoal` can also be used by other modded entities since it's not just a plain inner class.

Reply:
Alright i fixed this up. I also noticed another thing from bow attack goal to fix